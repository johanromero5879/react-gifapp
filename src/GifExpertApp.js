import { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
    const [ categories, setCategories ] = useState(defaultCategories )

    // const handleApp = () => {
    //     // setCategories( cats => [...cats, 'Kiss'] )
    //     setCategories([...categories, 'Kiss'])
        
    // }

    return <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={ setCategories } />
        <hr />
        { 
            categories.map( (category, index) => (
                <GifGrid 
                    category={ category }
                    key={ index } 
                />
            ) )
        }
    </>
}

export default GifExpertApp