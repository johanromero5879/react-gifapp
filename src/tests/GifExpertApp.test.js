import { shallow } from 'enzyme'

import GifExpertApp from '../GifExpertApp'

describe('Testing on <GifExpertApp />', () => {
    const categories = ['One punch', 'Kimetsu no Yaiba']

    test('should render properly', () => {
        const wrapper = shallow( <GifExpertApp /> )

        expect( wrapper ).toMatchSnapshot()
    })
    
    test('should show a categories list', () => {
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> )

        expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    })
    
})