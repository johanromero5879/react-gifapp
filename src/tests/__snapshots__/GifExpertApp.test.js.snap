// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <GifExpertApp /> should render properly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Testing on <GifExpertApp /> should show a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One punch"
    key="0"
  />
  <GifGrid
    category="Kimetsu no Yaiba"
    key="1"
  />
</Fragment>
`;
